pandas/src/parser/tokenizer.c

```diff
@@ -1255,6 +1242,35 @@ int parser_trim_buffers(parser_t *self) {
         }
     }
 
+    /* trim stream */
+    new_cap = _next_pow2(self->stream_len) + 1;
+    TRACE(("parser_trim_buffers: new_cap = %zu, stream_cap = %zu, lines_cap = %zu\n",
+           new_cap, self->stream_cap, self->lines_cap));
+    if (new_cap < self->stream_cap) {
+        TRACE(("parser_trim_buffers: new_cap < self->stream_cap, calling safe_realloc\n"));
+        newptr = safe_realloc((void*) self->stream, new_cap);
+        if (newptr == NULL) {
+            return PARSER_OUT_OF_MEMORY;
+        } else {
+            // realloc sets errno when moving buffer?
+            if (self->stream != newptr) {
```

Comment:
you have some extra comments in here. can you expand the explanation for future readers.